src/main/scala/net/snowflake/spark/snowflake/FilterPushdown.scala

```diff
@@ -63,24 +63,24 @@ private[snowflake] object FilterPushdown {
       dataType match {
```

Comment:
buildValueWithType() and buildValue() are internal functions. I believe many of the cases are not test-covered. I suggest you move out these 2 functions and make them as "private[snowflake]" and add unit test to cover them.

Reply:
The `FilterPushdown` object is already marked as `private[snowflake]`. These are 'static' functions.

Reply:
What I mean is these two functions are defined in function buildFilterStatement() not in class FilterPushdown. To test them with unit test, they should be moved out as FilterPushdown's member functions.

Reply:
So, I discovered that these functions require the variable `schema`, which is passed down through the top-level function `buildFilterStatement`, so I can't move these out of `buildFilterStatement` without refactoring. Since this is invoked from `SnowflakeRelation`, I would prefer to not perform extensive refactoring at this time.